README.md

```diff
@@ -27,3 +27,17 @@ about the current recursive call.
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. You do not need to prove that the invariant is correct. Add
 your answer to this markdown file.
+
+##### Answer
+
+The invariant is that, at the start of the recursive call, the return value 
```

Comment:
Don't quite understand what you mean here -- how can you know something about the return value at the start of the recursive call?